AU2/plugins/custom_plugins/PolicePlugin.py

```diff
@@ -88,122 +93,205 @@ def __init__(self):
                 "Police -> Set Ranks",
                 self.ask_set_ranks,
                 self.answer_set_ranks
+            ),
+            ConfigExport(
+                "PolicePlugin_toggle_auto",
+                "Police -> Toggle Auto Ranking",
+                lambda *args: [],
+                self.answer_toggle_auto
+            ),
+            ConfigExport(
+                "PolicePlugin_toggle_manual",
+                "Police -> Toggle Manual Ranking",
+                lambda *args: [],
+                self.answer_toggle_manual
+            ),
+            ConfigExport(
+                "PolicePlugin_toggle_police_kills_rankup",
+                "Police -> Toggle Ranking up on Police Kills",
+                lambda *args: [],
+                self.answer_toggle_police_kills_rankup
             )
         ]
 
-        self.ranks = GENERIC_STATE_DATABASE.arb_state.setdefault(self.identifier, DEFAULT_RANKS)
+    # I've done the database stuff in such a way that configs are only stored if changed; otherwise the hardcoded default configs are used.
+    # Not sure how sensible this is, but it seems to work
+
+    def gsdb_get(self, plugin_state_id):
+        return GENERIC_STATE_DATABASE.arb_state.get(self.identifier, {}).get(self.plugin_state[plugin_state_id]['id'], self.plugin_state[plugin_state_id]['default'])
 
-    def ask_set_ranks(self):  # TODO Make Ctrl-C work - not sure why it isn't
+    def gsdb_set(self, plugin_state_id, data):
+        GENERIC_STATE_DATABASE.arb_state.setdefault(self.identifier, {})[self.plugin_state[plugin_state_id]['id']] = data
+
+    def ask_set_ranks(self):
+        # TODO Make Ctrl-C work - not sure why it isn't --> Check Jamie's pull
+        default_text = self.gsdb_get("Ranks").copy()
+        default_text[int(self.gsdb_get("Default Rank"))] += "=default"
```

Comment:
very nit: You might consider checking that your default rank index does exist, and resorting to some sensible default otherwise (to avoid crashing in the event of an umpire fiddling with the DB).

Reply:
I've implemented this, but I'm not sure how worth it it is trying to prevent umpires breaking the system by manually editing the json poorly. There are so many instances of things that are expected to be ints or floats or datetimes that would break if a rogue umpire changes them in json to, say, strings; it would be a lot of effort to catch and fix every one of these potential issues, for very little gain